### DISCUSSION TITLE
Difference in WithEndpoint impl from preview 4 to preview 5

### DISCUSSION THREAD

**Discussion by maranmaran** [microsoft/aspire#3709] [Q&A] [answered]
Hi, I opened this issue today and it's closed. The proposed solution in comment works, but is different than what was working for me in preview 4 

https://github.com/dotnet/aspire/issues/3695

Can someone ELI5 to me why in preview 4 was this working for my case:

`.WithEndpoint(scheme: "http", hostPort: 5000, isProxied: false)`

I understand this was defining a endpoint that isn't proxied (ie Aspire doesn't generate any URL for it) and binds to host port (which I dynamically change in my service).

However, I don't have the same flexibility with new implementation of `WithEndpoint` in preview 5 ?
There's `port` and `targetPort`, but that doesn't seem to work unless I disable launch profile thing as per comment in GH Issue.

I presumed `WithExternalEndpoints` would work, and the code snippet in GH Issue shows it being set as External, but that doesn't work. I'm still curious as to why?

**Comment by davidfowl** [comment, marked as answer]
By default, when you add a project resource, we read your launch profile and add endpoints. Using this should still work fine:

```
.WithEndpoint(scheme: "http", hostPort: 5000, isProxied: false)
```

> I understand this was defining a endpoint that isn't proxied (ie Aspire doesn't generate any URL for it) and binds to host port (which I dynamically change in my service).

Non proxied endpoints means that you need to specify the port.

> However, I don't have the same flexibility with new implementation of WithEndpoint in preview 5 ?
There's port and targetPort, but that doesn't seem to work unless I disable launch profile thing as per comment in GH Issue.

If you don't disable the launch profile, we'll still end up using it in your appliction. That's the same as it was in preview 5.

> I presumed WithExternalEndpoints would work, and the code snippet in GH Issue shows it being set as External, but that doesn't work. I'm still curious as to why?

This doesn't do anything related to what you want, ignore it.


Looking at [your repository](
https://github.com/murh-lego/aspire-test/blob/5391ea43868d0c3643c8a4e842019295d6e8e211/WebApplication1/appsettings.json#L9-L19), you're defining endpoints in the kestrel configuration section. If you want to use these bindings then you do want to disable the launch profile and specify them in the apphost (as @mitchdenny said here https://github.com/dotnet/aspire/issues/3695#issuecomment-2056975023).

There's an open work item to support reading the kestrel configuration section for project resources to automatically define endpoints:

https://github.com/dotnet/aspire/issues/3456

**Reply by murh-lego** [reply]
Thank you for thorough explanation, I appreciate it 😸
